test/integration/confirmations/signatures/personalSign.test.tsx

```diff
@@ -0,0 +1,190 @@
+import * as backgroundConnection from '../../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventLocation,
+} from '../../../../shared/constants/metametrics';
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+
+jest.mock('../../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBakcgroundConnection = jest.mocked(backgroundConnection);
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const getMetaMaskStateWithUnapprovedPersonalSign = (accountAddress: string) => {
+  const pendingPersonalSignId = '0050d5b0-c023-11ee-a0cb-3390a510a0ab';
+  const pendingPersonalSignTime = new Date().getTime();
+  return {
+    ...mockMetaMaskState,
+    preferences: {
+      ...mockMetaMaskState.preferences,
+      redesignedConfirmationsEnabled: true,
+    },
+    unapprovedPersonalMsgs: {
+      [pendingPersonalSignId]: {
+        id: pendingPersonalSignId,
+        status: 'unapproved',
+        time: pendingPersonalSignTime,
+        type: MESSAGE_TYPE.PERSONAL_SIGN,
+        securityProviderResponse: null,
+        msgParams: {
+          from: accountAddress,
+          data: '0x4578616d706c652060706572736f6e616c5f7369676e60206d657373616765',
+          origin: 'https://metamask.github.io',
+          siwe: { isSIWEMessage: false, parsedMessage: null },
+        },
+      },
+    },
+    unapprovedPersonalMsgCount: 1,
+    pendingApprovals: {
+      [pendingPersonalSignId]: {
+        id: pendingPersonalSignId,
+        origin: 'origin',
+        time: pendingPersonalSignTime,
+        type: ApprovalType.PersonalSign,
+        requestData: {},
+        requestState: null,
+        expectsResult: false,
+      },
+    },
+    pendingApprovalCount: 1,
+  };
+};
+
+describe('PersonalSign Confirmation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('displays the header account modal with correct data', async () => {
+    const account =
+      mockMetaMaskState.internalAccounts.accounts[
+        mockMetaMaskState.internalAccounts
+          .selectedAccount as keyof typeof mockMetaMaskState.internalAccounts.accounts
+      ];
+
+    const accountName = account.metadata.name;
+    const mockedMetaMaskState = getMetaMaskStateWithUnapprovedPersonalSign(
+      account.address,
+    );
+
+    const { getByTestId, queryByTestId } = await integrationTestRender({
+      preloadedState: mockedMetaMaskState,
+      backgroundConnection: backgroundConnectionMocked,
+    });
+
+    expect(getByTestId('header-account-name')).toHaveTextContent(accountName);
+    expect(getByTestId('header-network-display-name')).toHaveTextContent(
+      'Chain 5',
+    );
+
+    fireEvent.click(getByTestId('header-info__account-details-button'));
+
+    await waitFor(() => {
```

Comment:
Nit: I haven't seen so many assertions in a `waitFor` callback before, is this typical? i.e. would you expect one of these elements to be found before the others had loaded?

I would expect the metric event might be slightly slower to show up, so that makes sense as an independent thing that we're waiting for. Though that one is a bit strange too, in that if the wrong metric event appeared first, we'd keep running failing assertions on it in a loop until we hit the timeout.

Maybe it'd make more sense to stage this as:
* Wait for account details modal
* Assert contents
* Wait for _some_ metric event to be captured
* Assert contents

Then we wouldn't have any redundant locators, and we'd fail faster if the wrong metric event appears first.

Reply:
I see that the same comment applies to a few other large `waitFor` callbacks in the other integration test module as well

Reply:
That makes sense. Done 👍